providers/FeedProvider.tsx

```diff
@@ -91,6 +96,41 @@ const FeedProvider = ({ children }: { children: ReactNode }) => {
     [feed, filter],
   );
 
+  const playFeedId = async (feedIndex: number) => {
+    const embed = findValidEmbed(feed[feedIndex]);
+    const feedObj = feed[feedIndex];
+    const url = embed?.url;
+    if (url) {
+      const metadata = await fetchMetadata(url, feedObj);
+      if (metadata)
+        dispatch({
+          type: 'PLAY',
+          payload: { metadata, feedId: feedObj.id },
+        });
+    }
+  };
+
+  const handleNext = async () => {
+    const feedIndex = feed.findIndex((feedObj: SupabasePost) => feedObj.id === player.feedId);
+
+    if (feedIndex > -1) {
+      if (feedIndex + 1 < feed.length) {
+        playFeedId(feedIndex + 1);
+      }
+    } else if (feedIndex <= -1) {
+      playFeedId(0);
+    }
+  };
+
+  const handlePrev = async () => {
+    const feedIndex = feed.findIndex((feedObj: SupabasePost) => feedObj.id === player.feedId);
+    if (feedIndex > -1) {
+      if (feedIndex && feedIndex > 0) {
+        playFeedId(feedIndex - 1);
+      }
+    }
```

Comment:
need `else` logic here, same as `handleNext`, in the case where `feedIndex <= -1`